File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.java
class Solution {
    public int solution(int[] num_list) {
        int cnt = 0;
        for (int n : num_list) {
            while (n != 1) {
                cnt++;
                if (n % 2 == 0) n /= 2;
                else n = (n - 1)/2;
            }
        }
        
        return cnt;
    }
}

File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.kt
class Solution {
    fun solution(num_list: IntArray): Int 
    = num_list.fold(0) { acc, num ->
        var cnt = 0
        var copy = num
        while (copy != 1) {
            cnt++
            copy /= 2
        }
        acc + cnt
    }
}